types/JSX.d.ts

```diff
@@ -0,0 +1,1955 @@
+//                 Benoit Benezech <https://github.com/bbenezech>
+//                 Patricio Zavolinsky <https://github.com/pzavolinsky>
+//                 Eric Anderson <https://github.com/ericanderson>
+//                 Dovydas Navickas <https://github.com/DovydasNavickas>
+//                 Josh Rutherford <https://github.com/theruther4d>
+//                 Guilherme Hübner <https://github.com/guilhermehubner>
+//                 Ferdy Budhidharma <https://github.com/ferdaber>
+//                 Johann Rakotoharisoa <https://github.com/jrakotoharisoa>
+//                 Olivier Pascal <https://github.com/pascaloliv>
+//                 Martin Hochel <https://github.com/hotell>
+//                 Frank Li <https://github.com/franklixuefei>
+//                 Jessica Franco <https://github.com/Jessidhia>
+//                 Saransh Kataria <https://github.com/saranshkataria>
+//                 Kanitkorn Sujautra <https://github.com/lukyth>
+//                 Sebastian Silbermann <https://github.com/eps1lon>
+//                 Kyle Scully <https://github.com/zieka>
+//                 Cong Zhang <https://github.com/dancerphil>
+//                 Dimitri Mitropoulos <https://github.com/dimitropoulos>
+//                 JongChan Choi <https://github.com/disjukr>
+//                 Victor Magalhães <https://github.com/vhfmag>
+//                 Dale Tan <https://github.com/hellatan>
+//                 Priyanshu Rav <https://github.com/priyanshurav>
+//                 Guilherme Correia <https://github.com/GuiDevloper>
+// TypeScript Version: 2.8
+
+import { NullstackClientContext } from ".";
+
+type NativeAnimationEvent = AnimationEvent;
+type NativeClipboardEvent = ClipboardEvent;
+type NativeCompositionEvent = CompositionEvent;
+type NativeDragEvent = DragEvent;
+type NativeFocusEvent = FocusEvent;
+type NativeKeyboardEvent = KeyboardEvent;
+type NativeMouseEvent = MouseEvent;
+type NativePointerEvent = PointerEvent;
+type NativeTransitionEvent = TransitionEvent;
+type NativeUIEvent = UIEvent;
+type NativeWheelEvent = WheelEvent;
+type Booleanish = boolean | 'true' | 'false';
+interface HTMLWebViewElement extends HTMLElement { }
+
+export interface ComponentLifecycle {
+    /**
+     * @see https://nullstack.app/full-stack-lifecycle#prepare
+     */
+    prepare?(context?: object): any
+
+    /**
+     * @see https://nullstack.app/full-stack-lifecycle#initiate
+     */
+    initiate?(context?: object): any
+
+    /**
+     * @see https://nullstack.app/full-stack-lifecycle#launch
+     */
+    launch?(context?: object): any
+
+    /**
+     * @see https://nullstack.app/full-stack-lifecycle#hydrate
+     */
+    hydrate?(context?: object): any
+
+    /**
+     * @see https://nullstack.app/full-stack-lifecycle#update
+     */
+    update?(context?: object): any
+
+    /**
+     * @see https://nullstack.app/full-stack-lifecycle#terminate
+     */
+    terminate?(context?: object): any
```

Comment:
```suggestion
    /**
     * @see https://nullstack.app/full-stack-lifecycle#prepare
     */
    prepare?(context?: Record<string, any>): any

    /**
     * @see https://nullstack.app/full-stack-lifecycle#initiate
     */
    initiate?(context?: Record<string, any>): any

    /**
     * @see https://nullstack.app/full-stack-lifecycle#launch
     */
    launch?(context?: Record<string, any>): any

    /**
     * @see https://nullstack.app/full-stack-lifecycle#hydrate
     */
    hydrate?(context?: Record<string, any>): any

    /**
     * @see https://nullstack.app/full-stack-lifecycle#update
     */
    update?(context?: Record<string, any>): any

    /**
     * @see https://nullstack.app/full-stack-lifecycle#terminate
     */
    terminate?(context?: Record<string, any>): any
```

Reply:
Maybe, if we know the type, use the correct type. `NullstackClientContext` or something?

Reply:
@brunolm We are using the `ComponentLifecycle` just as a model (and repetitive stuff, e.g. JSDocs) being really used/typed only in the Nullstack class at [`index.d.ts`](https://github.com/GuiDevloper/nullstack/blob/nullstack-jsx/types/index.d.ts#L33) (also avoiding duplicating in both files):
```ts
prepare?(context: NullstackContext & Props): any
initiate?(context: NullstackContext & Props): any
```
And because I had seen that the one implemented can't give typed methods.
Anyway, applied your better typing there!